### PR TITLE
Fix markdown whitespace issues in release notes template

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
This PR fixes a whitespace bug in `template.go` that produced invalid markdown. It was messing up the lists in the document and entries with multiline comments.

It also fixes a bug where `krel release-notes` produced json output when invoked with `--format=markdown` 

#### Does this PR introduce a user-facing change?

```release-note
Modified the release notes and changelog templates to fix rendering problems in headings, itemized lists and multiline comments.
Fixed a bug where `krel release-notes` produced json output when invoked with `--format=markdown` 
```

### CODE REVIEW

**Comment by puerco** [comment]
I was looking into the tests in the document package and something is weird. 

The release notes [markdown generated before using the template](https://raw.githubusercontent.com/kubernetes/sig-release/master/releases/release-1.18/release-notes-draft.md) did not contain a space before each entry, eg:
```markdown
#### kubeadm:
- command line option "kubelet-version" for `kubeadm upgrade node` has been deprecated and will be removed in a future release. ([#87942](https://github.com/kubernetes/kubernetes/pull/87942), [SataQiu](https://github.com/SataQiu)) [SIG Cluster Lifecycle]
- deprecate the usage of the experimental flag '--use-api' under the 'kubeadm alpha certs renew' command. ([#88827](https://github.com/kubernetes/kubernetes/pull/88827), [neolit123](https://github.com/neolit123)) [SIG Cluster Lifecycle]
- kube-dns is deprecated and will not be supported in a future version ([#86574](https://github.com/kubernetes/kubernetes/pull/86574), [SataQiu](https://github.com/SataQiu)) [SIG Cluster Lifecycle]
```
But now with the template, the space was introduced in the code and the golden files used to run the markdown tests contain a space before list items:
```markdown
### Deprecation
 - Deprecation #1.
 - This note is duplicated across SIGs.
```
This commit makes sure that the markdown output _does not_ contain a space before the generated list items and fixes the reference md files used for testing. 

I'm only concerned with the changelog because i'm note very familiar with that part. I don't know if these changes will mess something up there. Could you comment on that @saschagrunert  ?

**Comment by puerco** [comment]
> @puerco -- Thanks for the fix! Can you also ensure that each heading has an empty line following it?
> (That's another markdown lint fix that needs to be made in the template.)

Done @justaugustus headings now have a space

File: cmd/krel/cmd/changelog_test.go [hunk 1]
```diff
@@ -96,7 +96,6 @@ func TestNewPatchRelease(t *testing.T) { // nolint: dupl
 		require.Nil(t, err)
 		require.Contains(t, string(changelog), patchReleaseExpectedTOC)
 		require.Contains(t, string(changelog), patchReleaseExpectedContent)
-		require.Contains(t, string(changelog), patchReleaseDeps)
```
Comment (saschagrunert) [inline cmd/krel/cmd/changelog_test.go:99]: We still need this check, right?

**Comment by puerco** [comment]
> Thank you for the fix @puerco! The changelog uses an own template which would need adaptation there as well:
> https://github.com/kubernetes/release/blob/3544be5a6436265019b1e85606ba6b8e02d5ed48/cmd/krel/cmd/changelog.go#L156
> 
> Feel free to lift the hold if I should take care of that later on, because the integration tests needs to be changed as well if we change the template.
 
@saschagrunert  I've updated the changelog template and the integration tests as well. Took me a while but tests should be green now.

**Comment by puerco** [comment]
Oops, I accidentally deleted a test. Adding it back

File: cmd/krel/cmd/changelog_test.go (hunk 1 above)
Comment (puerco) [reply cmd/krel/cmd/changelog_test.go]: Yes sorry I had left it out when I was debugging. It's back now.

**Pull request merged**